app/controllers/file_manager_controller.rb

```diff
@@ -0,0 +1,240 @@
+# Code for file manager adapted from: adrientoub/file-explorer
+require 'archive'
+require 'pathname'
+
+class FileManagerController < ApplicationController
+  BASE_DIRECTORY = Rails.root.join('courses')
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :update_persistent_announcements
+
+  def index
+    check_path_exist('')
+    populate_directory(BASE_DIRECTORY, '')
+  end
+
+  def upload_index
+    upload_file('')
+    populate_directory(BASE_DIRECTORY, '')
+    render 'file_manager/index'
+  end
+
+  def path
+    absolute_path = check_path_exist(params[:path].nil? ? "" : params[:path])
+    if File.directory?(absolute_path) && check_instructor(absolute_path)
+      populate_directory(absolute_path, "#{params[:path]}/")
+      render 'file_manager/index'
+    elsif File.file?(absolute_path) && check_instructor(absolute_path)
+      if File.size(absolute_path) > 1_000_000 || params[:download]
+        send_file absolute_path
+      else
+        @file = File.read(absolute_path)
+        @path = params[:path]
+        @parent = './'
+        render :file, formats: :html
+      end
+    end
+  end
+
+  def upload
+    upload_file(params[:path].nil? ? "" : params[:path])
+  end
+
+  def delete
+    absolute_path = check_path_exist(params[:path])
+    if check_instructor(absolute_path)
+      if File.directory?(absolute_path)
+        FileUtils.rm_rf(absolute_path)
+      else
+        FileUtils.rm(absolute_path)
+      end
+    end
+    flash[:success] = "Successfully deleted"
+  end
+
+  def rename
+    absolute_path = check_path_exist(params[:relative_path])
+    if check_instructor(absolute_path)
+      current_path = Pathname.new(absolute_path)
+      parent = current_path.parent
+      if parent == BASE_DIRECTORY
+        flash[:error] = "Unable to rename courses in the root directory."
+      else
+        dir_name = File.dirname(params[:relative_path])
+
+        if params[:new_name].empty? || params[:new_name].nil?
+          raise ArgumentError, "New name not provided"
+        end
+
+        unless params[:new_name].match(/^[a-zA-Z0-9_-]+(\.[a-zA-Z0-9_-]+)?$/)
```

Comment:
Consider using `\A` and `\z` instead of `^` and `$` (see: https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions)

Reply:
Changed.